storage-api/src/java/org/apache/hadoop/hive/common/TableName.java

```diff
@@ -17,34 +17,45 @@
  */
 package org.apache.hadoop.hive.common;
 
+import java.io.Serializable;
+import java.util.Objects;
+
 /**
  * A container for a fully qualified table name, i.e. catalogname.databasename.tablename.  Also
  * includes utilities for string parsing.
  */
-public class TableName {
+public class TableName implements Serializable {
+
+  private static final long serialVersionUID = 1L;
+
+  /** Exception message thrown. */
+  private static final String ILL_ARG_EXCEPTION_MSG =
```

Comment:
I don't  really see the benefit of making this a static final string;
but anyway - I think the end goal is to stop the usage of any `tableName` which contains `dbname` or any other stuff...and in that case the method in which this is used should go away at some point - am I right?